### PR TITLE
net_if: add IFF_LOOPBACK/POINTOPOINT/MULTICAST/BROADCAST

### PR DESCRIPTION
**Pull request by zhhyu7**
VELAPLATFO-2660

Signed-off-by: zhanghongyu <zhanghongyu@xiaomi.com>
Change-Id: Ieb6b1eadefa8e2a60832346f71fff23c85d2d88a

### CODE REVIEW

File: include/nuttx/net/netdev.h [hunk 1]
```diff
@@ -242,7 +242,7 @@ struct net_driver_s
 
   /* Drivers interface flags.  See IFF_* definitions in include/net/if.h */
 
-  uint8_t d_flags;
+  uint32_t d_flags;
```
Comment (hartmannathan) [inline include/nuttx/net/netdev.h:245]: I just want to point out that in `struct ifreq` (declared in `include/net/if.h`), `ifru_flags` is defined as uint8_t. 

In `netdev_ifr_ioctl()` (in `net/netdev/netdev_ioctl.c`), when processing SIOCGIFFLAGS, we have this assignment:

```
      case SIOCGIFFLAGS:  /* Gets the interface flags */
        {
          dev = netdev_ifr_dev(req);
          if (dev)
            {
              req->ifr_flags = dev->d_flags;
            }

          ret = OK;
        }
        break;
```

So flags will be truncated.

Is this intentional?

**Comment by hartmannathan** [comment]
Please fix checkpatch error: "Remove Gerrit Change-ID's before submitting upstream" Thanks!

File: include/nuttx/net/netdev.h (hunk 1 above)
Comment (zhhyu7) [reply include/nuttx/net/netdev.h:245]: > 
Thank you for pointing this out.

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/net/if.h [hunk 2]
```diff
@@ -51,27 +51,43 @@
 #define IFF_RUNNING        (1 << 2) /* Carrier is available */
 #define IFF_IPv6           (1 << 3) /* Configured for IPv6 packet (vs ARP or IPv4) */
 #define IFF_BOUND          (1 << 4) /* Bound to a socket */
+#define IFF_LOOPBACK       (1 << 5) /* Is a loopback net */
+#define IFF_POINTOPOINT    (1 << 6) /* Is point-to-point link */
 #define IFF_NOARP          (1 << 7) /* ARP is not required for this packet */
+#define IFF_MULTICAST      (1 <<12) /* Supports multicast. */
+#define IFF_BROADCAST      (1 <<13) /* Broadcast address valid.  */
```
[the remark this answers is not in the document]
Comment (zhhyu7) [reply include/net/if.h]: > 

It has been corrected, Thanks

File: include/net/if.h [hunk 3]
```diff
@@ -51,27 +51,43 @@
 #define IFF_RUNNING        (1 << 2) /* Carrier is available */
 #define IFF_IPv6           (1 << 3) /* Configured for IPv6 packet (vs ARP or IPv4) */
 #define IFF_BOUND          (1 << 4) /* Bound to a socket */
+#define IFF_LOOPBACK       (1 << 5) /* Is a loopback net */
+#define IFF_POINTOPOINT    (1 << 6) /* Is point-to-point link */
 #define IFF_NOARP          (1 << 7) /* ARP is not required for this packet */
+#define IFF_MULTICAST      (1 <<12) /* Supports multicast. */
```
Comment (pkarashchenko) [inline include/net/if.h:57]: Maybe add extra space `1 << 12` and align comments accordingly?

**Review by pkarashchenko — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

File: include/net/if.h [hunk 4]
```diff
@@ -46,32 +46,48 @@
...
+#define IFF_DOWN           (1 << 0)  /* Interface is down */
+#define IFF_UP             (1 << 1)  /* Interface is up */
+#define IFF_RUNNING        (1 << 2)  /* Carrier is available */
+#define IFF_IPv6           (1 << 3)  /* Configured for IPv6 packet (vs ARP or IPv4) */
+#define IFF_BOUND          (1 << 4)  /* Bound to a socket */
+#define IFF_LOOPBACK       (1 << 5)  /* Is a loopback net */
+#define IFF_POINTOPOINT    (1 << 6)  /* Is point-to-point link */
+#define IFF_NOARP          (1 << 7)  /* ARP is not required for this packet */
+#define IFF_MULTICAST      (1 << 12) /* Supports multicast. */
+#define IFF_BROADCAST      (1 << 13) /* Broadcast address valid. */
```
Comment (pkarashchenko) [inline include/net/if.h:58]: Just a question: why 12 and 13 bits are used and not 8 and 9?

File: include/nuttx/net/netdev.h (hunk 1 above)
Comment (pkarashchenko) [inline include/nuttx/net/netdev.h:245]: Just a question: why type is expanded to 32bits and not to 16bits?

Comment (zhhyu7) [reply include/nuttx/net/netdev.h:245]: Refer to the definition of Linux, reserve space for further expansion

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/net/if.h (hunk 4 above)
Comment (pkarashchenko) [reply include/net/if.h:58]: This is still not answered.

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
@hartmannathan please review and merge if you are ok.

File: include/net/if.h (hunk 4 above)
Comment (zhhyu7) [reply include/net/if.h:58]: > Just a question: why 12 and 13 bits are used and not 8 and 9?

I was just reusing the Linux definition, do I need to change it to 8 and 9?

Comment (zhhyu7) [reply include/net/if.h:58]: > This is still not answered.

Sorry for my reply are not timely.

Comment (hartmannathan) [reply include/net/if.h:58]: Hello @zhhyu7

Sorry for my delay. I was away for a couple of days. The PR looks good. I will merge it now. If there is ever a need to change the bit values, they can always be changed.

**Review by hartmannathan — APPROVED** [review]
LGTM

**Pull request merged**